packages/gatsby/src/schema/schema.js

```diff
@@ -190,11 +191,29 @@ const updateSchemaComposer = async ({
     parentSpan: parentSpan,
   })
   activity.start()
-  await printTypeDefinitions({
-    config: printConfig,
-    schemaComposer,
-    parentSpan: activity.span,
-  })
+  if (!process.env.GATSBY_SKIP_WRITING_SCHEMA_TO_FILE) {
+    await printTypeDefinitions({
+      config: printConfig,
+      schemaComposer,
+      parentSpan: activity.span,
+    })
+    if (_CFLAGS_.GATSBY_MAJOR === `4` && shouldGenerateEngines()) {
+      // make sure to print schema that will be used when bundling graphql-engine
+      const graphqlEngineSnapshotPath = process.cwd() + `/.cache/schema.gql`
+      try {
+        await require(`fs-extra`).remove(graphqlEngineSnapshotPath)
```

Comment:
what's the reason for requiring lazy?

Reply:
That's on me - no real reason here, it should just be top level import - it was just marginally faster to iterate on code without jumping on code lines. We should make it regular top level import